### DISCUSSION TITLE
filter annotations by their title/text and get a list of similar names (like function in SQL)

### DISCUSSION THREAD

**Discussion by AhmedAredah** [pelicanmapping/osgearth#2586] [Support Q&A] [answered]
is there a direct way in OSGEarth to filter AnnotationNodes based on their text? and maybe zoom in to the bounding box of the filtered nodes?
Also, is there a way to change the color of the annotationNode icon or maybe highlight it?

Thank you.

**Comment by gwaldron** [comment, marked as answer]
There's no filtering of nodes in osgEarth; you will need to code that up yourself. You can use setNodeMask(0) to hide a node, and setNodeMask(~0) to show one. 

Perhaps the `ViewFitter` class will help with zooming to a region or node.

You can try adding a custom shader to an AnnotationNode to change its color at runtime, using the `VirtualProgram` construct.
